graph/cycle_check_undirected_graph.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file cycle_check_undirected graph.cpp
+ *
+ * @brief  [Disjoint Set (Or Union-Find) algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure) check for cycle in a undirected graph.
```

Comment:
Please add a detailed description and a better brief description (see, for example, the [`n_queens`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/backtracking/n_queens.cpp) algorithm).